src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,155 @@
     public T get(int index) {
-        return null;
+        checkIndex(index, (size == 0) ? 0 : (size - 1));
+        return elements(index);
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index, (size == 0) ? 0 : (size - 1));
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index, (size == 0) ? 0 : (size - 1));
+        final Object[] es = elements;
+
+        @SuppressWarnings("unchecked") T oldValue = (T) es[index];
+        final int newSize;
+        if ((newSize = size - 1) > index) {
+            System.arraycopy(es, index + 1, es, index, newSize - index);
+        }
+        es[size = newSize] = null;
+        return oldValue;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        Object[] es = elements;
+
+        for (int i = 0; i < this.size; i++) {
+
+            if (equal(element, (T) es[i])) {
+                remove(i);
+                return element;
+            }
+        }
+        throw new NoSuchElementException("There is no such element.");
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Object[] grow(int minCapacity) {
+        int oldCapacity = elements.length;
+        if (oldCapacity > 0 || elements != DEFAULT_ELEMENTS) {
+            int newCapacity = newLength(oldCapacity, minCapacity - oldCapacity,
+                    oldCapacity >> 1);
+            return elements = Arrays.copyOf(elements, newCapacity);
+        } else {
+            return elements = new Object[Math.max(DEFAULT_CAPACITY, minCapacity)];
+        }
+    }
+
+    private int newLength(int oldLength, int minGrowth, int prefGrowth) {
+        int prefLength = oldLength + Math.max(minGrowth, prefGrowth);
+        if (0 < prefLength && prefLength <= SOFT_MAX_ARRAY_LENGTH) {
+            return prefLength;
+        } else {
+            return hugeLength(oldLength, minGrowth);
+        }
+    }
+
+    private int hugeLength(int oldLength, int minGrowth) {
+        int minLength = oldLength + minGrowth;
+        if (minLength < 0) {
+            throw new OutOfMemoryError(
+                    "Required array length " + oldLength + " + "
+                            + minGrowth + " is too large");
+        } else {
+            return Math.max(minLength, SOFT_MAX_ARRAY_LENGTH);
+        }
+    }
+
+    private void checkIndex(int index, int size) {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Index passed to the method is negative or "
+                            + "bigger than size of ArrayList. "
+                            + "Index: " + index + ", Size: " + size);
+        }
     }
+
+    private boolean equal(T element1, T element2) {
```

Comment:
remove this method and use its logic in the place where you need it, also rewrite your logic so you won't need ifs